driver/ppm_fillers.c

```diff
@@ -4201,9 +4203,31 @@ int f_sys_procexit_e(struct event_filler_arguments *args)
 	 * status
 	 */
 #ifndef UDIG
-	res = val_to_ring(args, args->sched_prev->exit_code, 0, false, 0);
+	/* Exit status */
+	res = val_to_ring(args, __WEXITSTATUS(args->sched_prev->exit_code), 0, false, 0);
```

Comment:
Won't this change the status param? I agree that it's awkward to force higher-level code to mask the exit value, but some higher level code might be already doing that, and changing the param would break that higher level code.

Reply:
You are right! I can propose to add a "ret" param as second one, so that we will have 4 params:
* status that is exit status exactly same as kernel side, to retain backward compatibility
* ret that is real return code
* sig that is the signal that killed the process
* core true false if a coredump was created

What do you think?

Reply:
In latest commit i implemented the requested change :)

Reply:
That makes a lot more sense to me, thanks for doing it.